server/src/controllers/eventController.service.ts

```diff
@@ -40,67 +37,109 @@ export const createStripeCheckoutSession = async (
     metadata: {
       sessionType: '__ticketing',
       contactID,
-      donation,
       discountCode: null,
     },
-    ...(couponID && {discounts: [{coupon: couponID}]}),
+    ...(discount.code != '' && discount.amountOff && {discounts: [{coupon: (await createStripeCoupon(discount)).id}]}),
   };
+
   const session = await stripe.checkout.sessions.create(checkoutObject);
   return {id: session.id};
 };
 
-export const createStripeCoupon = async (discount: any, orderTotal: number) => {
-  const amountOff = getDiscountAmount(discount, orderTotal);
+export const expireCheckoutSession = async (
+    id: string,
+) => stripe.checkout.sessions.expire(id);
 
-  const stripeCoupon = await stripe.coupons.create({
-    [discount.amount ? 'amount_off' : 'percent_off']: discount.amount ?
-      amountOff * 100 :
-      discount.percent,
+export const createStripeCoupon = async (discount: any) => {
+  return stripe.coupons.create({
+    ['amount_off']: discount.amountOff*100,
     duration: 'once',
     name: discount.code,
     currency: 'usd',
   });
-  return stripeCoupon.id;
 };
 
-export const getOrderItems = async (
+export const getDonationItems = (donationCartItems: number[]) => {
+  const donations: any[] = [];
+  const donationCartRows: LineItem[] = [];
+  let donationTotal = 0;
+  donationCartItems.forEach((donation) => {
+    if (donation < 0) {
+      throw new Error('Amount of donation can not be negative');
+    } else if (!donation) {
+      return;
+    }
+    donations.push({
+      amount: donation,
+      frequency: freq.one_time,
+    });
+    donationCartRows.push(
+        getCartRow('Donation', 'A generous donation', donation*100, 1),
+    );
+    donationTotal+=donation;
+  });
+
+  return {donations, donationCartRows, donationTotal};
+};
```

Comment:
Is there a time where there could be more than one donation?

Reply:
Updated.